File: LUA-noparallel/noparallel.lua
#!/usr/bin/luajit

globi = 3000000
primct = 0

function primtest()
    while true do
	local i=globi
	globi=globi-1

	if i < 2 then
	    return -- end
	end
	local prim = true
	local sqi = math.floor(math.sqrt(i))
	for j=2,sqi do
	    if i%j == 0 then
		prim=False
		break
	    end
	end
	if prim == true then
	    primct=primct+1
	end
    end
end

primtest()
print (primct)

File: Python-noparallel/noparallel.py
#!/usr/bin/pypy
# -*- coding: utf8 -*-

MAXNUM = 3000000

import math

if __name__ == '__main__':
    i = MAXNUM
    primct = 0
    for i in xrange(MAXNUM, 1, -1):
	prim = True
	sqi = int(math.sqrt(i))
	for j in xrange(2, sqi+1):
	    if not (i%j):
		prim=False
		break
	if prim == True:
	    primct+=1

    print primct
